consensus/misc/rewards.go

```diff
@@ -62,3 +72,39 @@ func FindMinDenominations(reward *big.Int) map[uint8]uint8 {
 
 	return denominationCount
 }
+
+// calculateExchangeRate calculates the exchange rate for the block
+func CalculateExchangeRate(parent *types.WorkObject) (*big.Int, *big.Int, *big.Int, error) {
+	qiToQuai := parent.Header().QiToQuai() //Units of Qi
+	quaiToQi := parent.Header().QuaiToQi() //Units of Quai
+
+	//Calculate the new amounts of qi and quai being exchanged
+	for _, etx := range parent.ExtTransactions() {
+		if etx.Conversion() {
+			if etx.To().IsInQuaiLedgerScope() {
+				qiToQuai = new(big.Int).Add(etx.Value(), qiToQuai)
```

Comment:
> qiToQuai => Units of Qi
But here we add units of Quai (inQuaiLedgerScope).

So I'm confused if this is correct or not. If its correct could you make the comments more clear or add one on this line to explain?

Reply:
added comments